utils/cputils.py

```diff
@@ -0,0 +1,153 @@
+"""
+These collection of functions runs CellProfiler and renames the .sqlite outputs to any specified name if 
+running an analysis pipeline.
+"""
+
+# must use the annotations import as CellProfiler is restricted to Python 3.8 at this time so Optional
+# by itself only works in Python 3.10
+from __future__ import annotations
+from typing import Optional
+import os
+import subprocess
+import pathlib
+
+
+def rename_sqlite_file(sqlite_dir_path: pathlib.Path, name: str):
+    """Rename the .sqlite file into {name}.sqlite as to differentiate between different files.
+
+    Args:
+        sqlite_dir_path (pathlib.Path): path to CellProfiler_output directory
+        name (str): new name for the SQLite file
+
+    Raises:
+        FileNotFoundError: This error will occur if you do not have a SQLite file with the hard coded file name in the specified directory.
+        This means that this function can not find the right file to rename, so it raises an error.
+    """
+    try:
+        # CellProfiler requires a name to be set in to pipeline, so regardless of plate or method, all sqlite files name are hardcoded
+        sqlite_file_path = pathlib.Path(f"{sqlite_dir_path}/NF1_data.sqlite")
+
+        new_file_name = str(sqlite_file_path).replace(
+            sqlite_file_path.name, f"{name}.sqlite"
+        )
+
+        # change the file name in the directory
+        pathlib.Path(sqlite_file_path).rename(pathlib.Path(new_file_name))
+        print(f"The file is renamed to {pathlib.Path(new_file_name).name}!")
+
+    except FileNotFoundError as e:
+        print(
+            f"The NF1_data.sqlite file is not found in directory. Either the pipeline wasn't ran properly or the file is already renamed.\n"
+            f"{e}"
+        )
+
+
+def run_cellprofiler(
+    path_to_pipeline: str,
+    path_to_output: str,
+    path_to_images: str,
+    sqlite_name: Optional[None | str] = None,
+    analysis_run: Optional[False | bool] = False,
+):
+    """Run CellProfiler on data. It can be used for either a illumination correction pipeline (default) and analysis pipeline (when
+    parameter is set to True).
+
+    Args:
+        path_to_pipeline (str): path to the CellProfiler .cppipe file with the segmentation and feature measurement modules
+        path_to_output (str): path to the output folder
+        path_to_images (str): path to the images
+        sqlite_name (str, optional): string with name for SQLite file for an analysis pipeline if you plan on running
+        multiple sets of images (e.g., per plate) so that the outputs will have different names (default is None)
+        analysis_run (bool, optional): will use functions to complete an analysis pipeline (default is False)
+    """
+    # check to make sure paths to pipeline and directory of images are correct before running the pipeline
+    if not pathlib.Path(path_to_pipeline):
```

Comment:
This line of code is prone to bugs. For example, if you provide a string path that may not exist, then this `if` statement will always be set to `True`. 

Based on the error message below. It seems you're trying to see if the `path_to_pipeline` is a file ? correct me if I'm wrong. If my assumption is correct, consider applying these changes.
```suggestion
    if not pathlib.Path(path_to_pipeline).is_file():
```
This will return `True` if the path is a file, else it will return false.

Reply:
Further reading your code, it seems if you're trying to check if the file exists overall. 

a better change would be this:
```suggestion
    if not pathlib.Path(path_to_pipeline).exist():
```
or you can reduce code by using `resolve()`
```suggestion
    if not pathlib.Path(path_to_pipeline).resolve(strict=True):
```
where `strict=True` will check if the file exists before generating an absolute path. 

To me it seems `exist()` is the most appropriate here because you're not using path,you're just checking if it exists.

Reply:
You are correct in your assumption! I have added the `resolve(strict=True)` option so that it checks to see if the file exists which it needs to exist for CellProfiler to work.